src/Components/Shared/src/WebEventData/DragEventArgsReader.cs

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Text.Json;
+
+namespace Microsoft.AspNetCore.Components.Web
+{
+    internal static class DragEventArgsReader
+    {
+        private static readonly JsonEncodedText DataTransfer = JsonEncodedText.Encode("dataTransfer");
+        private static readonly JsonEncodedText DropEffect = JsonEncodedText.Encode("dropEffect");
+        private static readonly JsonEncodedText EffectAllowed = JsonEncodedText.Encode("effectAllowed");
+        private static readonly JsonEncodedText Files = JsonEncodedText.Encode("files");
+        private static readonly JsonEncodedText Items = JsonEncodedText.Encode("items");
+        private static readonly JsonEncodedText Types = JsonEncodedText.Encode("types");
+        private static readonly JsonEncodedText Kind = JsonEncodedText.Encode("kind");
+        private static readonly JsonEncodedText Type = JsonEncodedText.Encode("type");
```

Comment:
Do you happen to know whether the static constructor runs (for all the types) during app startup, or is it deferred until the first time we see an event of the corresponding type?

If it's the latter that's great. If it all happens on startup, do you think in the wasm case all the string encoding would have any perceptible effect on startup time?

Reply:
The type constructor will run the first time the type is needed/used. It might be in startup if for some reason this gets referenced from the startup code path, otherwise, likely the first time we receive an event via JS interop, since likely mono will trigger the class load.

Reply:
There’s one big event-handling method that references all of these types in different branches. I’m unsure whether that’s enough to trigger all their static constructors at once, or if it’s only when we enter the branches that reference each one.

Reply:
TBH since it’s really on the first event, not during actual app startup before first UI render, I’m sure it will be fine to initialise them all then anyway.

Marking this as resolved.